sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/SupportsDelete.java

```diff
@@ -70,6 +77,34 @@ default boolean canDeleteWhere(Filter[] filters) {
    */
   void deleteWhere(Filter[] filters);
 
+  default boolean canDeleteWhere(Predicate[] predicates) {
+    List<Filter> filterList = new ArrayList();
+    for (int i = 0; i < predicates.length; i++) {
```

Comment:
I prefer to use enhanced `for` loop

Reply:
+1

Reply:
I suggest using Java 8 lambda expression.

Reply:
> I suggest using Java 8 lambda expression.

+1, If there is no strong performance demand

Reply:
Thanks for the comments. I added a toV1 method that takes Array of Filters in the Util class to eliminate the duplicated code.